R/utilities-qualification.R

```diff
@@ -202,3 +202,65 @@ getOutputsFromGOFMergedPlotsConfiguration <- function(plot) {
 # Vector of allowable plot types in configuration plan.
 # TODO deprecate vector using enum ConfigurationPlots
 QualificationPlotTypes <- c("GOFMergedPlots", "ComparisonTimeProfilePlots", "DDIRatioPlots", "TimeProfile")
+
+
+
+
+#' @title separateVariableFromUnit
+#' @description Split a string containing a varialbe and unit description into a list of two strings that separates the variable and the unit.
+#' @param variableUnitString is a string that has the format 'Variable [unit]' or 'Variable'
+#' @return A named list, with fields 'name' and 'unit'.  If variableUnitString lacks a '[unit]' substring, then the unit field is returned as an empty string.
+separateVariableFromUnit <- function(variableUnitString) {
+  splitVariableUnitString <- strsplit(x = sub("[ []", replacement = "", x = variableUnitString), split = "[][]")[[1]]
+  return(list(
+    name = trimws(splitVariableUnitString[1]),
+    unit = ifelse(test = is.na(x = splitVariableUnitString[2]), yes = "", no = splitVariableUnitString[2])
+  ))
+}
+
+
+#' @title parseObservationsDataFrame
+#' @description Function to read the variable names and units in the first two columns of an observations data frame used for qualification. First column stores timepoints.  Second column stores measurements of a quantity corresponding to timepoints in first column.
+#' @param observationsDataFram
+#' @return A named list with `time` and `output` fields.  Each field contains a list that is output by `separateVariableFromUnit` with fields that store the variable name and the unit.
+parseObservationsDataFrame <- function(observationsDataFrame) {

```

Comment:
@abdullahhamadeh , there are some cases in which there is a 3rd column for uncertainty.
Should we include a 3rd field for uncertainty in that list ?

Reply:
Done